app/src/main/java/com/epfl/beatlink/ui/profile/ChangePassword.kt

```diff
@@ -0,0 +1,178 @@
+      topBar = { ScreenTopAppBar("Change password", "changePasswordTitle", navigationActions) },
+      bottomBar = {
+        BottomNavigationMenu(
+            onTabSelect = { route -> navigationActions.navigateTo(route) },
+            tabList = LIST_TOP_LEVEL_DESTINATION,
+            selectedItem = navigationActions.currentRoute())
+      }) { paddingValues ->
+        Column(
+            modifier =
+                Modifier.padding(paddingValues)
+                    .padding(16.dp)
+                    .verticalScroll(rememberScrollState()),
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              Text(
+                  text = "Please enter your existing\npassword and your new password.",
+                  style = MaterialTheme.typography.displayLarge,
+                  color = MaterialTheme.colorScheme.primary,
+                  modifier =
+                      Modifier.fillMaxWidth().padding(bottom = 15.dp).testTag("instructionText"))
+
+              // Current Password input field
+              CustomInputField(
+                  value = currentPassword,
+                  onValueChange = { currentPassword = it },
+                  label = "Current Password",
+                  placeholder = "Enter password",
+                  keyboardType = KeyboardType.Password,
+                  visualTransformation = PasswordVisualTransformation(),
+                  modifier = Modifier.testTag("inputCurrentPassword"))
+
+              // New Password input field
+              CustomInputField(
+                  value = newPassword,
+                  onValueChange = { newPassword = it },
+                  label = "New Password",
+                  placeholder = "Enter password",
+                  keyboardType = KeyboardType.Password,
+                  visualTransformation = PasswordVisualTransformation(),
+                  supportingText = "6-18 characters",
+                  modifier = Modifier.testTag("inputNewPassword"))
+
+              // Confirm New Password input field
+              CustomInputField(
+                  value = confirmNewPassword,
+                  onValueChange = { confirmNewPassword = it },
+                  label = "Confirm New Password",
+                  placeholder = "Enter password",
+                  keyboardType = KeyboardType.Password,
+                  visualTransformation = PasswordVisualTransformation(),
+                  supportingText = "6-18 characters",
+                  modifier = Modifier.testTag("inputConfirmNewPassword"))
+
+              Spacer(modifier = Modifier.height(16.dp))
+
+              ChangePasswordButton(
+                  authViewModel = firebaseAuthViewModel,
+                  currentPassword = currentPassword,
+                  newPassword = newPassword,
+                  confirmNewPassword = confirmNewPassword)
+            }
+      }
+}
+
+@Composable
```

Comment:
Instead of re-creating a composable you should use the `PrincipalButton` composable in `MainComponents` and put your logic inside of the `onClick` parameter

Reply:
Thanks for you comment, I changed it.